src/lib/y2partitioner/actions/create_partition_table.rb

```diff
@@ -0,0 +1,98 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2partitioner/actions/transaction_wizard"
+require "y2partitioner/actions/controllers"
+require "y2partitioner/dialogs"
+
+module Y2Partitioner
+  module Actions
+    # Action for creating a new partition table
+    class CreatePartitionTable < TransactionWizard
+      attr_reader :controller
+
+      # @param disk_name [String]
+      def initialize(disk_name)
+        textdomain "storage"
+
+        super()
+        @controller = Controllers::PartitionTable.new(disk_name)
+      end
+
+    protected
+
+      # @see TransactionWizard
+      def sequence_hash
+        {
+          "ws_start"    => "select_type",
+          "select_type" => { next: "confirm" },
+          "confirm"     => { next: "commit" },
+          "commit"      => { finish: :finish }
+        }
```

Comment:
No, because there might not be a first dialog because there might be only one choice. Think of S/390 and DASDs. Other architectures might also not offer both MS-DOS and GPT. In that case, we still want the confirmation, but no partition type selection.